DataManagementSystem/DB/FileCatalogComponents/DirectoryTreeBase.py

```diff
@@ -336,148 +337,156 @@ def getDirectoryParameters( self, path ):
 #####################################################################
   def _setDirectoryParameter( self, path, pname, pvalue ):
     """ Set a numerical directory parameter
+
+        :param mixed path: Directory path or paths as a string or directory ID as int,
+                           list/tuple of ints or a string to select directory IDs
+        :param str pname: parameter name
+        :param int pvalue: parameter value
     """
-    result = self.__getDirID( path )
-    if not result['OK']:
-      return result
-    dirID = result['Value']
-    req = "UPDATE FC_DirectoryInfo SET %s=%d WHERE DirID=%d" % ( pname, pvalue, dirID )
+    result = getIDSelectString( path )
+    if not result['OK'] and isinstance( path, basestring ):
+      result = self.__getDirID( path )
+      if not result['OK']:
+        return result
+      dirID = result['Value']
+      result = getIDSelectString( dirID )
+      if not result['OK']:
+        return result
+
+    dirIDString = result['Value']
+    req = "UPDATE FC_DirectoryInfo SET %s=%d, " \
```

Comment:
Strange that my comment here disappeared from #2554. I wrote that I plan to split the DirectoryLevelTree into a generic directory info part and support for the tree structure. At this point it will be appropriate to make this rearrangement